invenio_records_rest/views.py

```diff
@@ -941,6 +953,60 @@ def check_etag(self, etag, weak=False):
                     )
 
 
+class SearchAsYouTypeResource(MethodView):
+    """Resource for `search_as_you_type` searches."""
+
+    view_name = '{0}_search_as_you_type'
+
+    def __init__(self, search_as_you_type, search_class=None, **kwargs):
+        """Constructor."""
+        self.search_as_you_type = search_as_you_type
+        self.search_as_you_type_fields = set(self.search_as_you_type.keys())
+        self.search_class = search_class
+
+    def get(self, **kwargs):
+        """Get results from `search_as_you_type`."""
+        request_fields = set(request.values.keys())
+        key = self.search_as_you_type_fields & request_fields
```

Comment:
I would change naming there cause it's not clear to me (is the size of it always 1? if not, why are you taking only one key from it?)

Reply:
the `key` should always be `1`, what would you suggest for a name?

Reply:
Anything different than key and indicating that what is that set, maybe `matched_search_as_you_type_fields`